src/prefect/tasks/prefect/flow_run.py

```diff
@@ -114,12 +123,19 @@ def run(
         # grab the ID for the most recent version
         flow_id = flow[0].id
 
+        if is_hosted_backend and context.get("flow_run_id"):
```

Comment:
The upcoming server release will support idempotency keys so I'm going to open an issue to revert this trap

Reply:
I can add a TODO also